# HG changeset patch
# User Jaroslav Hajek <highegg@gmail.com>
# Date 1231514711 -3600
#      Fri Jan 09 16:25:11 2009 +0100
# Node ID 9e1973f7709d3ea437743c3dc5a951c0fa6061e5
# Parent  d6a349c7bd394775df34949f55baa87574bd454f
fix missing widening ops for diag & perm matrices

diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,17 @@
+2009-01-09  Jaroslav Hajek  <highegg@gmail.com>
+
+	* OPERATORS/op-cdm-cdm.cc: Install missing widening op.
+	* OPERATORS/op-dm-dm.cc: Dtto.
+	* OPERATORS/op-fcdm-fcdm.cc: Dtto.
+	* OPERATORS/op-fdm-fdm.cc: Dtto.
+	* OPERATORS/op-pm-pm.cc: Dtto.
+	* OPERATORS/op-fpm-fpm.cc: Dtto.
+
 2009-01-09  Jaroslav Hajek  <highegg@gmail.com>
 
 	* ov-cell.cc (octave_cell::subsasgn): Put erased copy back after
 	make_unique has been called.
 
 2009-01-09  Kai Habel <kai.habel@gmx.de>
 
         * graphics.cc: Calculate normals for surface boundaries, use
diff --git a/src/OPERATORS/op-cdm-cdm.cc b/src/OPERATORS/op-cdm-cdm.cc
--- a/src/OPERATORS/op-cdm-cdm.cc
+++ b/src/OPERATORS/op-cdm-cdm.cc
@@ -101,9 +101,10 @@ install_cdm_cdm_ops (void)
   INSTALL_BINOP (op_mul, octave_complex_diag_matrix, octave_complex_diag_matrix, mul);
   INSTALL_BINOP (op_div, octave_complex_diag_matrix, octave_complex_diag_matrix, div);
   INSTALL_BINOP (op_ldiv, octave_complex_diag_matrix, octave_complex_diag_matrix, ldiv);
 
   INSTALL_CONVOP (octave_complex_diag_matrix, octave_complex_matrix, complex_diag_matrix_to_complex_matrix);
   INSTALL_CONVOP (octave_complex_diag_matrix, octave_float_complex_diag_matrix, 
                   complex_diag_matrix_to_float_complex_diag_matrix);
   INSTALL_ASSIGNCONV (octave_complex_diag_matrix, octave_complex_matrix, octave_complex_matrix);
+  INSTALL_WIDENOP (octave_complex_diag_matrix, octave_complex_matrix, complex_diag_matrix_to_complex_matrix);
 }
diff --git a/src/OPERATORS/op-dm-dm.cc b/src/OPERATORS/op-dm-dm.cc
--- a/src/OPERATORS/op-dm-dm.cc
+++ b/src/OPERATORS/op-dm-dm.cc
@@ -94,9 +94,10 @@ install_dm_dm_ops (void)
   INSTALL_BINOP (op_sub, octave_diag_matrix, octave_diag_matrix, sub);
   INSTALL_BINOP (op_mul, octave_diag_matrix, octave_diag_matrix, mul);
   INSTALL_BINOP (op_div, octave_diag_matrix, octave_diag_matrix, div);
   INSTALL_BINOP (op_ldiv, octave_diag_matrix, octave_diag_matrix, ldiv);
 
   INSTALL_CONVOP (octave_diag_matrix, octave_matrix, diag_matrix_to_matrix);
   INSTALL_CONVOP (octave_diag_matrix, octave_float_diag_matrix, diag_matrix_to_float_diag_matrix);
   INSTALL_ASSIGNCONV (octave_diag_matrix, octave_matrix, octave_matrix);
+  INSTALL_WIDENOP (octave_diag_matrix, octave_matrix, diag_matrix_to_matrix);
 }
diff --git a/src/OPERATORS/op-fcdm-fcdm.cc b/src/OPERATORS/op-fcdm-fcdm.cc
--- a/src/OPERATORS/op-fcdm-fcdm.cc
+++ b/src/OPERATORS/op-fcdm-fcdm.cc
@@ -97,13 +97,16 @@ install_fcdm_fcdm_ops (void)
   INSTALL_UNOP (op_hermitian, octave_float_complex_diag_matrix, hermitian);
 
   INSTALL_BINOP (op_add, octave_float_complex_diag_matrix, octave_float_complex_diag_matrix, add);
   INSTALL_BINOP (op_sub, octave_float_complex_diag_matrix, octave_float_complex_diag_matrix, sub);
   INSTALL_BINOP (op_mul, octave_float_complex_diag_matrix, octave_float_complex_diag_matrix, mul);
   INSTALL_BINOP (op_div, octave_float_complex_diag_matrix, octave_float_complex_diag_matrix, div);
   INSTALL_BINOP (op_ldiv, octave_float_complex_diag_matrix, octave_float_complex_diag_matrix, ldiv);
 
-  INSTALL_CONVOP (octave_float_complex_diag_matrix, octave_complex_diag_matrix, float_complex_diag_matrix_to_complex_diag_matrix);
+  INSTALL_CONVOP (octave_float_complex_diag_matrix, octave_complex_diag_matrix, 
+                  float_complex_diag_matrix_to_complex_diag_matrix);
   INSTALL_CONVOP (octave_float_complex_diag_matrix, octave_float_complex_matrix, 
                   float_complex_diag_matrix_to_float_complex_matrix);
   INSTALL_ASSIGNCONV (octave_float_complex_diag_matrix, octave_float_complex_matrix, octave_float_complex_matrix);
+  INSTALL_WIDENOP (octave_float_complex_diag_matrix, octave_complex_diag_matrix, 
+                   float_complex_diag_matrix_to_complex_diag_matrix);
 }
diff --git a/src/OPERATORS/op-fdm-fdm.cc b/src/OPERATORS/op-fdm-fdm.cc
--- a/src/OPERATORS/op-fdm-fdm.cc
+++ b/src/OPERATORS/op-fdm-fdm.cc
@@ -94,9 +94,10 @@ install_fdm_fdm_ops (void)
   INSTALL_BINOP (op_sub, octave_float_diag_matrix, octave_float_diag_matrix, sub);
   INSTALL_BINOP (op_mul, octave_float_diag_matrix, octave_float_diag_matrix, mul);
   INSTALL_BINOP (op_div, octave_float_diag_matrix, octave_float_diag_matrix, div);
   INSTALL_BINOP (op_ldiv, octave_float_diag_matrix, octave_float_diag_matrix, ldiv);
 
   INSTALL_CONVOP (octave_float_diag_matrix, octave_float_matrix, float_diag_matrix_to_float_matrix);
   INSTALL_CONVOP (octave_float_diag_matrix, octave_diag_matrix, float_diag_matrix_to_diag_matrix);
   INSTALL_ASSIGNCONV (octave_float_diag_matrix, octave_float_matrix, octave_float_matrix);
+  INSTALL_WIDENOP (octave_float_diag_matrix, octave_float_matrix, float_diag_matrix_to_float_matrix);
 }
diff --git a/src/OPERATORS/op-fpm-fpm.cc b/src/OPERATORS/op-fpm-fpm.cc
--- a/src/OPERATORS/op-fpm-fpm.cc
+++ b/src/OPERATORS/op-fpm-fpm.cc
@@ -82,10 +82,11 @@ install_fpm_fpm_ops (void)
 
   INSTALL_BINOP (op_mul, octave_float_perm_matrix, octave_float_perm_matrix, mul);
   INSTALL_BINOP (op_div, octave_float_perm_matrix, octave_float_perm_matrix, div);
   INSTALL_BINOP (op_ldiv, octave_float_perm_matrix, octave_float_perm_matrix, ldiv);
 
   INSTALL_CONVOP (octave_float_perm_matrix, octave_float_matrix, float_perm_matrix_to_float_matrix);
   INSTALL_CONVOP (octave_float_perm_matrix, octave_perm_matrix, float_perm_matrix_to_perm_matrix);
   INSTALL_ASSIGNCONV (octave_float_perm_matrix, octave_float_matrix, octave_float_matrix);
+  INSTALL_WIDENOP (octave_float_perm_matrix, octave_float_matrix, float_perm_matrix_to_float_matrix);
 }
 
diff --git a/src/OPERATORS/op-pm-pm.cc b/src/OPERATORS/op-pm-pm.cc
--- a/src/OPERATORS/op-pm-pm.cc
+++ b/src/OPERATORS/op-pm-pm.cc
@@ -77,9 +77,10 @@ install_pm_pm_ops (void)
 
   INSTALL_BINOP (op_mul, octave_perm_matrix, octave_perm_matrix, mul);
   INSTALL_BINOP (op_div, octave_perm_matrix, octave_perm_matrix, div);
   INSTALL_BINOP (op_ldiv, octave_perm_matrix, octave_perm_matrix, ldiv);
 
   INSTALL_CONVOP (octave_perm_matrix, octave_matrix, perm_matrix_to_matrix);
   INSTALL_CONVOP (octave_perm_matrix, octave_float_perm_matrix, perm_matrix_to_float_perm_matrix);
   INSTALL_ASSIGNCONV (octave_perm_matrix, octave_matrix, octave_matrix);
+  INSTALL_WIDENOP (octave_perm_matrix, octave_matrix, perm_matrix_to_matrix);
 }
